core/query/sql.ml

```diff
@@ -137,138 +152,163 @@ let order_by_clause n =
    returned. This allows these operators to take lists that have any
    element type at all. *)
 
-let rec string_of_query quote ignore_fields q =
-  let sq = string_of_query quote ignore_fields in
-  let sb = string_of_base quote false in
-  let sbt = string_of_base quote true in
-  let string_of_fields fields =
+let pp_constant ppf s = Format.fprintf ppf "%s" s
+let pp_comma ppf () = Format.fprintf ppf ","
+
+let rec pr_query ppf quote ignore_fields q =
+  let pp_quote ppf q = Format.fprintf ppf "%s" (quote q) in
+
+  let pr_q ppf q = pr_query ppf quote ignore_fields q in
+  let pr_b ppf q = pr_base ppf quote false q in
+  let pr_b_ignore_fields ppf q = pr_base ppf quote true q in
+
+  let gen_pp_pair ppf fmt_str fl fr (l, r) = Format.fprintf ppf fmt_str fl l fr r in
+  let gen_pp_option ppf fmt_str f option =
+    match option with
+      | None -> ()
+      | Some x -> Format.fprintf ppf fmt_str f x
+  in
+
+  let pr_fields ppf fields =
+    let pp_field ppf (b, l) = gen_pp_pair ppf "(%a) as %a" pr_b pp_constant (b, l) in
     if ignore_fields then
-      "0 as \"@unit@\"" (* SQL doesn't support empty records! *)
+      Format.fprintf ppf "%a" pp_constant "0 as \"@unit@\"" (* SQL doesn't support empty records! *)
```

Comment:
This is equivalent to `pp_constant ppf "0 as \"@unit@\""` which is in turn equivalent to `Format.pp_print_string ppf "0 as \"@unit@\""`.